bundles/org.openhab.core.semantics/src/main/java/org/openhab/core/semantics/Properties.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.semantics;
+
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * This class provides a stream of all defined properties.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+@NonNullByDefault
+public class Properties {
+
+    static final Set<Class<? extends Property>> PROPERTIES = ConcurrentHashMap.newKeySet();
+
+    static {
+        PROPERTIES.add(Property.class);
+    }
+
+    public static Stream<Class<? extends Property>> stream() {
+        return PROPERTIES.stream();
+    }
+
+    public static boolean add(Class<? extends Property> tag) {
```

Comment:
I am not sure what is purpose of Properties, Locations and Points classes after your changes, can you elaborate?

Reply:
Nothing changes here except that these classes were previously generated in model.
I just moved the classes.
Their purposes remain the same as before, that is everything is either a location, an equipment, a point or a property.

Reply:
In fact, these 4 classes are apparently not used !
They could be probably removed.

Reply:
Removed.